service/s3/s3manager/upload.go

```diff
@@ -508,6 +508,7 @@ func (u *multiuploader) upload(firstBuf io.ReadSeeker) (*UploadOutput, error) {
 
 	// Read and queue the rest of the parts
 	for u.geterr() == nil {
+		num++
```

Comment:
The previous change would increment to 3 and then return an `io.EOF` since there are no more parts. In this case it would start at 1, increment to 2 and upload the next part with no errors. Finally, increment to three and return the excess error, but we want the `io. ErrUnexpectedEOF`. Does that sound right?